QUARKUS-1079.md

```diff
@@ -0,0 +1,73 @@
+# QUARKUS-1079 - Hibernate Reactive as Tech Preview
+
+JIRA link: https://issues.redhat.com/browse/QUARKUS-1079
+
+For native mode similar test coverage as for JVM mode needs to be ensured.
+
+## Scope of the testing
+This feature added previews of two extensions: 
+ - Hibernate Reactive
+ - Hibernate Reactive with Panache
+
+Following test coverage should be implemented for Hibernate Reactive:
+- Usage of application.properties file for passing reactive-hibernate settings
+- Usage of import.sql
+- Smoke test(getting a working Mutiny session)
+- Retrieving Multi result from the DB
+- Retrieving Uni result(require a transformation) from the DB
+- Usage of DB-generated ids and ids generated with Java code(via ReactiveIdentifierGenerator)
+- Validation of values
+- Attribute converters
+- Usage of CompletionStages alongside a Mutiny API
+- Usage of session opened through openSession() method as well as withSession()  
+- Usage of transaction started through openSession() method as well as withSession()
+- Usage of session and transaction opened with the same method
+- Session methods: find, persist, remove, merge, flush
+- Queries: regular, native, named, JPA criteria
+- Query methods: setParameter, setMaxResults, setFirstResult, getSingleResult, getResultList, executeUpdate
+- Eager and lazy fetches
+- Field-level lazy fetching(Hibernate team "doesn't encourage" using this feature, though)
+- Stateless sessions and their methods
+ 
+Following test coverage should be implemented for Hibernate Reactive with Panache:
+- Active record pattern
+- Repository pattern
+- list and stream methods
+- find method
+- sorting 
+- Named queries
+- DTO projection
+- Transactions
+- Id generation
+
+Ideally, all of aforementioned tests should run for PostgreSQL, MySQL and DB2 connections.
+
+### Impact on testsuites and testing automation:
+ - Ensure this coverage works both in JVM and NATIVE mode
+ - Tests were created in "examples" section of quarkus-test-framework, maybe they can be moved to quarkus-test-suite
+
+### Impact on resources:
+ - More services will be running on OpenShift cluster, impact on CPU/memory should be minimal
```

Comment:
We should probably run a (sub)set of the scenarios on bare metal as well, with test containers for DB provisioning

Reply:
I added considerations for bare metal tests.